promise.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+	let retrieve = document.getElementById('retrieve');
+    
+    const getResult = function() {
+      if (xhr.readyState === 4) {
```

Comment:
I'm pretty sure this would have crashed when run, correct? Can you explain why I have this hunch?

Reply:
Because you wanted to create a promise. You wanted to make your code more readable and easier to understand.

Reply:
Will crush because in the getResult function I have no an event called onreadystatechange which verifies the state of 4.